app/dl_formats.py

```diff
@@ -60,12 +57,14 @@ def get_opts(format: str, quality: str, ytdl_opts: dict) -> dict:
     if "postprocessors" not in opts:
         opts["postprocessors"] = []
 
-    if format == "mp3":
+    if format in ("m4a", "mp3"):
         opts["postprocessors"].append({
             "key": "FFmpegExtractAudio",
-            "preferredcodec": "mp3",
+            "preferredcodec": format,
             "preferredquality": 0 if quality == "best" else quality,
         })
+    
+    if format in ("m4a", "mp3"):
```

Comment:
why the additional "if" statement? seems like it's identical to the one above, so it's not necessary